sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1215,4 +1215,129 @@ class SparkSqlAstBuilder extends AstBuilder {
+
+    val syntaxTypeErrorStr = if (createPipelineDatasetHeaderCtx.materializedView() != null) {
+      "MATERIALIZED VIEW"
+    } else if (createPipelineDatasetHeaderCtx.streamingTable() != null) {
+      "STREAMING TABLE"
+    } else {
+      // Should never be possible based on grammar definition.
+      throw invalidStatement(ctx.getText, ctx)
+    }
+
+    val ifNotExists = createPipelineDatasetHeaderCtx.EXISTS() != null
+    val provider = Option(ctx.tableProvider).map(_.multipartIdentifier.getText)
+    val (colDefs, colConstraints) = Option(ctx.tableElementList()).map(visitTableElementList)
+      .getOrElse((Nil, Nil))
+
+    if (colConstraints.nonEmpty) {
+      throw operationNotAllowed("Pipeline datasets do not currently support column constraints. " +
+        "Please remove and CHECK, UNIQUE, PK, and FK constraints specified on the pipeline " +
+        "dataset.", ctx)
+    }
+
+    val (partTransforms, partCols, bucketSpec,
+    properties, options, location, comment, collation, serdeInfoOpt,
+    clusterBySpec) = visitCreateTableClauses(ctx.createTableClauses())
+
+    val partitioning =
+      partitionExpressions(partTransforms, partCols, ctx) ++
+        clusterBySpec.map(_.asTransform)
+
+    // Because the createTableClauses grammar is reused for createPipelineDataset but pipeline
+    // datasets don't support bucketing, options, storage location, or Hive SerDe, validate they
+    // are not set.
+    if (bucketSpec.isDefined) {
+      throw operationNotAllowed(s"Bucketing is not supported for CREATE $syntaxTypeErrorStr " +
+        "statements. Please remove any bucket spec specified in the statement.", ctx)
+    }
+    if (options.options.nonEmpty) {
+      throw operationNotAllowed(s"Options are not supported for CREATE $syntaxTypeErrorStr " +
+        "statements. Please remove any OPTIONS lists specified in the statement.", ctx)
+    }
+    serdeInfoOpt.map(serdeInfo => if (serdeInfo.storedAs.nonEmpty) {
+      throw operationNotAllowed(s"The STORED AS syntax is not supported for CREATE " +
+        s"$syntaxTypeErrorStr statements. Consider using the Data Source based USING clause "
+        + "instead.", ctx)
+    } else {
+      throw operationNotAllowed(s"Hive SerDe format options are not supported for CREATE " +
+        s"$syntaxTypeErrorStr statements.", ctx)
+    })
+    if (location.nonEmpty) {
+      throw operationNotAllowed(s"Specifying location is not supported for CREATE " +
+        s"$syntaxTypeErrorStr statements. The storage location for a pipeline dataset is " +
+        "managed by the pipeline itself.", ctx)
+    }
+
+    val spec = TableSpec(
+      properties = properties,
+      provider = provider,
+      options = Map.empty,
+      location = location,
+      comment = comment,
+      collation = collation,
+      serde = None,
+      external = false,
+      constraints = Seq.empty
+    )
+
+    withIdentClause(createPipelineDatasetHeaderCtx.identifierReference, ident => {
```

Comment:
Given the `CreatePipelineDataset#name` field is a `LogicalPlan`, we can just do
```
CreateMaterializedViewAsSelect
  name = withIdentClause(...),
  ...)
```
instead of
```
withIdentClause... {
  CreateMaterializedViewAsSelect(...)
}
```